src/dos/drive_virtual.cpp

```diff
@@ -181,13 +193,14 @@ void VFILE_Register(const char *name,
 	auto new_file  = std::make_shared<VFILE_Block>();
 	new_file->name = vfilenames[vfile_pos].shortname;
 	vfile_pos++;
-	new_file->data = data ? new (std::nothrow) uint8_t[size] : nullptr;
-	if (new_file->data)
-		memcpy(new_file->data, data, size);
-	new_file->size = size;
-	new_file->date = fztime || fzdate ? fzdate : default_date;
-	new_file->time = fztime || fzdate ? fztime : default_time;
-	new_file->onpos = onpos;
+	new_file->data = std::make_shared<std::vector<uint8_t>>();
+	if (data) {
+		new_file->data->resize(size);
+		memcpy(&((*new_file->data)[0]), data, size);
+	}
```

Comment:
`vector`s can read C arrays at contruction time, so this will help avoid the memcpy compexity:

```suggestion
if (data && size > 0) {
    new_file->data = std::make_shared<std::vector<uint8_t>>(data, data + size);
} else {
    new_file->data = std::make_shared<std::vector<uint8_t>>();
}
```

Reply:
Hmm.. maybe the `else` case isn't even needed and we can leave the shared pointer empty?

I guess we'd need calling code to check `if (!data || data.empty())` and skip.

Reply:
OK, changed.